From d0ca9da6245712d40a0ffc0c34e86d8227b8aea1 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 2 Jul 2018 18:41:16 +0800
Subject: [PATCH 4274/5242] MLK-19017-6 drm/imx: add system pm support for SEC
 DSIM

commit  9b8b88065df6ab55bb97206859252129dc414dd0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Implement the suspend()/resume() callbacks to support system
power management functions for SEC DSIM.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit db3e9faa0278af6de5aaac008478123d0ebecb73)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/sec_mipi_dsim-imx.c |   24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/drivers/gpu/drm/imx/sec_mipi_dsim-imx.c b/drivers/gpu/drm/imx/sec_mipi_dsim-imx.c
index 4edff1c..e57451d 100644
--- a/drivers/gpu/drm/imx/sec_mipi_dsim-imx.c
+++ b/drivers/gpu/drm/imx/sec_mipi_dsim-imx.c
@@ -55,6 +55,20 @@ struct imx_sec_dsim_device {
 
 static struct imx_sec_dsim_device *dsim_dev;
 
+#if CONFIG_PM
+static int imx_sec_dsim_runtime_suspend(struct device *dev);
+static int imx_sec_dsim_runtime_resume(struct device *dev);
+#else
+static int imx_sec_dsim_runtime_suspend(struct device *dev)
+{
+	return 0;
+}
+static int imx_sec_dsim_runtime_resume(struct device *dev)
+{
+	return 0;
+}
+#endif
+
 static void disp_mix_dsim_soft_reset_release(struct regmap *gpr, bool release)
 {
 	if (release)
@@ -252,6 +266,16 @@ static int imx_sec_dsim_remove(struct platform_device *pdev)
 #ifdef CONFIG_PM_SLEEP
 static int imx_sec_dsim_suspend(struct device *dev)
 {
+	return imx_sec_dsim_runtime_suspend(dev);
+}
+
+static int imx_sec_dsim_resume(struct device *dev)
+{
+	return imx_sec_dsim_runtime_resume(dev);
+}
+#else
+static int imx_sec_dsim_suspend(struct device *dev)
+{
 	return 0;
 }
 
-- 
1.7.9.5

